tests/components/tedee/test_init.py

```diff
@@ -67,3 +67,38 @@ async def test_bridge_device(
     )
     assert device
     assert device == snapshot
+
+
+async def test_cleanup_disconnected_locks(
+    hass: HomeAssistant,
+    mock_config_entry: MockConfigEntry,
+    mock_tedee: MagicMock,
+    device_registry: dr.DeviceRegistry,
+    entity_registry: er.EntityRegistry,
+    snapshot: SnapshotAssertion,
+) -> None:
+    """Ensure disconnected locks are cleaned up."""
+    mock_config_entry.add_to_hass(hass)
+    await hass.config_entries.async_setup(mock_config_entry.entry_id)
+    await hass.async_block_till_done()
+
+    def assert_current_registry_state() -> None:
+        devices = dr.async_entries_for_config_entry(
+            device_registry, mock_config_entry.entry_id
+        )
+        assert devices == snapshot
```

Comment:
Instead of using snapshots here, wouldn't it make more sense to just assert the amount of devices and if the device we expect is removed is actually removed?

I like snapshot tests, but only if we need to test the contents of big data structures. Not if we need to check if something is missing from a list.